src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -128,90 +142,104 @@ public void RelayWorker()
         {
             this.logger.LogTrace("()");
 
-            this.asyncLoop = this.asyncLoopFactory.Run($"{this.name}.RelayWorker", async token =>
+            this.relayWorkerTask = Task.Factory.StartNew(async delegate
             {
-                this.logger.LogTrace("()");
-
-                int announceBlockCount = this.blocksToAnnounce.Count;
-                if (announceBlockCount == 0)
+                while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
                 {
-                    this.logger.LogTrace("(-)[NO_BLOCKS]");
-                    return;
-                }
+                    // Wait until a new block is added to the queue.
+                    this.blockEnqueued.Wait();
 
-                this.logger.LogTrace("There are {0} blocks in the announce queue.", announceBlockCount);
+                    // Make sure that at least 'FlushFrequencySeconds' seconds passed since the last announcement.
+                    // This is needed in order to ensure announcing blocks in batches to reduce the overhead.
+                    double secondsSinceLastAnnounce = (DateTime.Now - this.lastBlockAnnounceTimeStamp).TotalSeconds;
```

Comment:
int here and milliseconds

Reply:
we don't use date functions directly, always use IDateTimeProvider

Reply:
Fixed for the sake of code consistency.